messaging/src/main/java/org/axonframework/messaging/configuration/MessageHandlingComponent.java

```diff
@@ -0,0 +1,151 @@
+     *
+     * @param name    The name the given {@code handler} can handle.
+     * @param handler The handler instance that handles {@link org.axonframework.messaging.Message messages} for the
+     *                given name.
+     * @return This registry for fluent interfacing.
+     */
+    default MessageHandlingComponent subscribe(@Nonnull QualifiedName name,
+                                               @Nonnull MessageHandler handler) {
+        switch (handler) {
+            case MessageHandlingComponent component:
+                component.supportedCommands().forEach(command -> subscribe(command, (CommandHandler) component));
+                component.supportedEvents().forEach(command -> subscribe(command, (EventHandler) component));
+                component.supportedQueries().forEach(command -> subscribe(command, (QueryHandler) component));
+                break;
+            case CommandHandler commandHandler:
+                subscribe(name, commandHandler);
+                break;
+            case EventHandler eventHandler:
+                subscribe(name, eventHandler);
+                break;
+            case QueryHandler queryHandler:
+                subscribe(name, queryHandler);
+                break;
+            default:
+                throw new IllegalStateException("Unexpected value: " + handler);
+        }
+        return this;
+    }
+
+    @Override
+    MessageHandlingComponent subscribe(@Nonnull QualifiedName name,
+                                       @Nonnull CommandHandler commandHandler);
+
+    default MessageHandlingComponent subscribe(@Nonnull Set<QualifiedName> names,
+                                               @Nonnull CommandHandler commandHandler) {
+        names.forEach(name -> subscribe(name, commandHandler));
+        return this;
+    }
+
+    @Override
+    MessageHandlingComponent subscribe(@Nonnull QualifiedName name, @Nonnull EventHandler eventHandler);
+
+    @Override
+    default MessageHandlingComponent subscribe(@Nonnull Set<QualifiedName> names,
+                                               @Nonnull EventHandler eventHandler) {
+        names.forEach(name -> subscribe(name, eventHandler));
+        return this;
+    }
+
+    @Override
+    MessageHandlingComponent subscribe(@Nonnull QualifiedName name, @Nonnull QueryHandler queryHandler);
+
+    @Override
+    default MessageHandlingComponent subscribe(@Nonnull Set<QualifiedName> names,
+                                               @Nonnull QueryHandler queryHandler) {
+        names.forEach(name -> subscribe(name, queryHandler));
+        return this;
+    }
+
+    /**
+     * All supported {@link org.axonframework.messaging.Message messages}, referenced through a {@link QualifiedName}.
+     *
+     * @return All supported {@link org.axonframework.messaging.Message messages}, referenced through a
+     * {@link QualifiedName}.
+     */
+    default Set<QualifiedName> supportedMessages() {
```

Comment:
Something to think about for our future API. I know this is just a playground function. 

As messages are treated very differently in behaviour, I think we should avoid grouping the names of different types together. Users should be specific in which message type they are looking for the supported messages for.
I am afraid we will see people accidentally registering all queries to a command bus, for example :D

Reply:
There's definitely something to be said about this... Let's be on the safe side and remove this method.